### DISCUSSION TITLE
Fungsi difftime

### DISCUSSION THREAD

**Discussion by joejnt** [teamalgoritma/community#846] [Q&A] [answered]
Dear Tim Algoritma,

Mohon bantuannya jika saya ingin menghitung rentang waktu dari data seperti dibawah ini bagaimana caranya ya ?

![image](https://github.com/teamalgoritma/community/assets/103427989/d3634805-bb69-4aab-8f17-ef0e1f417f25)

Mohon saran dan petunjuknya,

Best regards,
JT

**Comment by IrfanChairurrachman** [comment]
Dear @joejnt ,

Terima kasih atas pertanyaannya, 
Untuk rentang waktu yang dimaksud ingin dihitungnya seperti apa ya Pak dan berharap outputnya seperti apa ya Pak?

Terima kasih

**Reply by joejnt** [reply]
Dear @IrfanChairurrachman 

Hasil yang diharapkan, saya bisa menghitung rentang waktu jam transaksi mesin atm 168010 jam 10:42:15 ke 10:42:35 dan itu dikelompokkan per hari per mesin atm.

Best regards,
JT

**Reply by IrfanChairurrachman** [reply, marked as answer]
Dear @joejnt ,

Untuk mencari perbedaan waktu yang dikelompokkan berdasarkan id (group) dan hari, dapat menggunakan `groupby()`, kemudian mengurutkan waktu dan terakhir menghitung perbedaan waktu menggunakan `diff()`. Seperti contoh kode di bawah ini

```{R}
library(dplyr)
library(lubridate)

# Contoh data
df <- data.frame(
  time = c("2024-03-05 10:00:00", "2024-03-06 11:00:00", "2024-03-06 12:00:00",
           "2024-03-05 10:15:00", "2024-03-06 11:30:00", "2024-03-06 12:45:00"),
  group = c("A", "A", "A", "B", "B", "B")
)

# Ubah tipe data
df$time <- ymd_hms(df$time)
# Buat kolom hari
df$date <- date(df$time)

# Group by menggunakan group
df <- df %>%
  group_by(group, date) %>%
  arrange(time) %>%
  mutate(time_diff = c(NA, diff(time))) %>%
  arrange(group) %>% 
  ungroup()

df
```

Dengan konsep yang sama, semoga kode ini dapat membantu untuk menjawab pertanyaan Anda.

Demikian jawaban yang dapat kami sampaikan. Semoga membantu! Jika ternyata jawaban ini menyelesaikan permasalahan Anda, silakan mark as anwer.

**Reply by joejnt** [reply]
Dear @IrfanChairurrachman 

Well noted,

Thx u

Best regards,
JT
